index.ts

```diff
@@ -0,0 +1,152 @@
+  if (!isString(name)) return null;
+
+  return (
+    (json || []).find(function (country: CountryRaw) {
+      if (useAlias) {
+        return (
+          country.name.toUpperCase() === name.toUpperCase() ||
+          country.alpha2.toUpperCase() === name.toUpperCase() ||
+          (country.alias || []).find(function (alias) {
+            return alias.toUpperCase() === name.toUpperCase();
+          })
+        );
+      }
+      return (
+        country.name.toUpperCase() === name.toUpperCase() ||
+        country.alpha2.toUpperCase() === name.toUpperCase()
+      );
+    }) ?? null
+  );
+}
+
+export const findCountryByName = getCountryByName;
+export const findCountryByNameOrShortName = getCountryByNameOrShortName;
+
+/**
+ * Find the province object of the given province name
+ *
+ * @param {String}  name              english province name
+ * @param {Boolean} [useAlias]        use alias flag, default `false`
+ * @return {Object} province          province object
+ */
+function getProvinceByName(
+  this: CountryRaw,
+  name?: Maybe<string>,
+  useAlias?: Maybe<boolean>,
+): Maybe<Province> {
+  if (!isString(name) || !Array.isArray(this.provinces)) return null;
+
+  return (
+    (this.provinces || []).find(function (province: Province) {
+      if (useAlias) {
+        return (
+          province.name.toUpperCase() === name.toUpperCase() ||
+          (province.alias || []).find(function (alias) {
+            return alias.toUpperCase() === name.toUpperCase();
+          })
+        );
+      }
+      return province.name.toUpperCase() === name.toUpperCase();
+    }) ?? null
+  );
+}
+
+/**
+ * Find the province object of the given province name or short name
+ *
+ * @param {String}  name              english province name or short name
+ * @param {Boolean} [useAlias]        use alias flag, default `false`
+ * @return {Object} province          province object
+ */
+function getProvinceByNameOrShortName(
+  this: CountryRaw,
+  name?: Maybe<string>,
+  useAlias?: Maybe<boolean>,
+): Maybe<Province> {
+  if (!isString(name) || !Array.isArray(this.provinces)) return null;
+
+  return (
+    (this.provinces || []).find(function (province: Province) {
+      if (useAlias) {
+        return (
+          province.name.toUpperCase() === name.toUpperCase() ||
+          (province.short && province.short.toUpperCase() === name.toUpperCase()) ||
+          (province.alias || []).find(function (alias) {
+            return alias.toUpperCase() === name.toUpperCase();
+          })
+        );
+      }
+      return (
+        province.name.toUpperCase() === name.toUpperCase() ||
+        (province.short && province.short.toUpperCase() === name.toUpperCase())
+      );
+    }) ?? null
+  );
+}
+
+/**
+ *
+ * Add search function to each country and map each country by alpha2
+ */
+const listCountries = keyBy(cloneDeep(countriesRaw), 'alpha2');
+const countries: { [countryCode: string]: Country } = Object.keys(listCountries).reduce(
```

Comment:
This means any script importing the library will run this `reduce` on all countries. Is this an API that you need to support for backwards compatibility? Couldn't it be changed to something like `getProvinceByName(countries.US)` ?

Reply:
it was for backwards compatibility but you're right it's better to remove it

Reply:
done